### PR TITLE
Add expected tests for previews

### PR DESCRIPTION
**Pull request by komalali**

### CODE REVIEW

File: environment.go [hunk 1]
```diff
@@ -88,3 +88,15 @@ func (e *Environment) GetTemporaryFiles() map[string]Value {
...
+	props := e.Properties
+	json := make(map[string]any, len(props))
+	if props == nil {
+		return nil
+	}
+	for k, v := range props {
+		json[k] = v.ToJSON(true)
+	}
+	return json
+}
```
Comment (pgavlin) [inline environment.go:102]: Can we remove this method and just call `NewValue(e.Properties).ToJSON(true)`?

File: eval/eval_test.go [hunk 2]
```diff
@@ -218,6 +218,7 @@ func TestEval(t *testing.T) {
 		CheckDiags  syntax.Diagnostics `json:"checkDiags,omitempty"`
 		EvalDiags   syntax.Diagnostics `json:"evalDiags,omitempty"`
 		Environment *esc.Environment   `json:"environment,omitempty"`
+		Preview     map[string]any     `json:"preview,omitempty"`
```
Comment (pgavlin) [inline eval/eval_test.go:221]:
```suggestion
		CheckJSON     map[string]any     `json:"checkJson,omitempty"`
```

Comment (pgavlin) [reply eval/eval_test.go:221]: (it would also be nice to have `EvalJSON` as a parallel, but we can do that as a follow-up)

Comment (komalali) [reply eval/eval_test.go]: fixed and added EvalJSON too

File: eval/eval_test.go [hunk 3]
```diff
@@ -236,18 +238,24 @@ func TestEval(t *testing.T) {
...
+
 				bytes, err := json.MarshalIndent(expectedData{
 					LoadDiags:   loadDiags,
 					CheckDiags:  checkDiags,
 					EvalDiags:   evalDiags,
 					Environment: actual,
+					EvalJSON:    evalJSON,
+					CheckJSON:   checkJSON,
 				}, "", "    ")
+				bytes = append(bytes, '\n')
```
Comment (pgavlin) [inline eval/eval_test.go:258]: just curious why we need this bit

**Review by pgavlin — APPROVED** [review]

File: eval/eval_test.go (hunk 3 above)
Comment (komalali) [reply eval/eval_test.go:258]: coz I hate this icon
Screenshot 2023-11-07 at 1 51 00 PM

**Pull request merged**